changehc/delphi_changehc/download_ftp_files.py

```diff
@@ -0,0 +1,71 @@
+"""
+Downloads files modified in the last 24 hours from the specified ftp server."""
+
+# standard
+import datetime
+import functools
+import sys
+from os import path
+
+# third party
+import paramiko
+
+class AllowAnythingPolicy(paramiko.MissingHostKeyPolicy):
+    def missing_host_key(self, client, hostname, key):
+        return
+
+
+def print_callback(filename, bytes_so_far, bytes_total):
+    rough_percent_transferred = int(100 * (bytes_so_far / bytes_total))
+    if (rough_percent_transferred % 25) == 0:
+        print(f'{filename} transfer: {rough_percent_transferred}%')
+
+
+def get_files_from_dir(sftp, out_path):
+    current_time = datetime.datetime.now()
+    seconds_in_day = 24 * 60 * 60
+
+    # go through files in recieving dir
+    files_to_download = []
+    for fileattr in sftp.listdir_attr():
+        file_time = datetime.datetime.fromtimestamp(fileattr.st_mtime)
+        time_diff_to_current_time = current_time - file_time
+        if time_diff_to_current_time.total_seconds() <= seconds_in_day:
+            files_to_download.append(fileattr.filename)
+
+    filepaths_to_download = {}
+    for file in files_to_download:
```

Comment:
Is this checking if we already have the file? low pri, but for brevity this could get added to the end of the above loop where it checks the <24 hr condition, so it becomes
```
for files in sftp
    if <24hrs old and not exists already
        add to list to download
```

Reply:
Changed